lib/ui/serverlessFunctionLogs.ts

```diff
@@ -82,20 +108,32 @@ function processLog(log, options) {
   }
 }
 
-function processLogs(logsResp, options) {
-  if (!logsResp || (logsResp.results && !logsResp.results.length)) {
-    return 'No logs found.';
-  } else if (logsResp.results && logsResp.results.length) {
-    return logsResp.results
-      .map(log => {
+function isLogsResponse(
+  logsResp: LogsResponse | Log
+): logsResp is LogsResponse {
+  return (
+    'results' in logsResp &&
+    Array.isArray(logsResp.results) &&
+    logsResp.results !== undefined &&
+    logsResp.results.length > 0
+  );
+}
+
+function processLogs(
+  logsResp: LogsResponse | Log,
+  options: Options
+): string | void {
+  if (isLogsResponse(logsResp)) {
+    return logsResp
+      .results!.map(log => {
         return processLog(log, options);
       })
       .join('\n');
   }
-  return processLog(logsResp, options);
+  return processLog(logsResp as Log, options);
```

Comment:
Is this type assertion necessary?  The options are `LogsReponse | Log` and return in the if statement if it is a `LogsResponse`